### PR TITLE
jsonschema: add filesystem loader for use with embed

### PR DESCRIPTION
**Pull request by julesjcraske**
```
//go:embed *.json
var fs embed.FS
var schema = jsonschema.NewFS(fs).LoadJSONExt
...
zs.Register("view_subscriptions", "2019-11-03", schema("view_subscriptions"), rpc.ViewSubscriptions)
```

### CODE REVIEW

**Review by billinghamj — COMMENTED** [review]
If this was intended for very generic use, I would say this is good. But given we need it to make a specific part of crpc extremely concise, I don't think this is really the way to go, or at least we could wrap it further in crpc - will add an example branch off this one to show what that could look like

File: lib/jsonschema/filesystem.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+// NewFS will typically be passed an embed.FS to easily load separate .json schema files
+func NewFS(raw fs.FS) *FS {
+	return &FS{raw: raw}
+}
+
+// Loads the given filename suffixing with ".json"
+func (f *FS) LoadJSONExt(filepath string) gojsonschema.JSONLoader {
+	return f.Load(fmt.Sprintf("%s.json", filepath))
```
Comment (billinghamj) [inline lib/jsonschema/filesystem.go:22]: json schemas _have_ to be JSON, so I'm not sure there's much benefit to allowing this as an option - would just make it the only way

**Comment by billinghamj** [comment]
Actually I take it back!

I was going to suggest this:

```go
func SchemaLoader(fs fs.FS) func(string) gojsonschema.JSONLoader {
	js := jsonschema.NewFS(fs)

	return func(name string) gojsonschema.JSONLoader {
		return js.LoadJSONExt(name)
	}
}
```

But seeing how you've used it here:

```go
//go:embed *.json
var fs embed.FS
var schema = jsonschema.NewFS(fs).LoadJSONExt
```

It is both generic/decoupled and perfectly concise! 👏 Definitely had not thought of setting the function on the variable

**Review by billinghamj — APPROVED** [review]

**Review by mfiedorowicz — APPROVED** [review]
Like it!

**Review by Duffleman — APPROVED** [review]

**Pull request merged**